src/classes/SftpSessionHandler.ts

```diff
@@ -741,19 +756,36 @@ export class SftpSessionHandler {
           case 'xa+': // append and read (file must not exist)
           case 'a': // append
           {
-            const temporaryFile = tmp.fileSync();
-            this.openTemporaryFiles.set(handle, {
-              ...temporaryFile,
-              path: filePath,
+            tmp.file((err, name, fd, removeCallback) => {
```

Comment:
Just a note that these changes LOOK large but in reality I just moved all the previous logic into the callback.